lib/IRGen/GenDecl.cpp

```diff
@@ -2734,9 +2734,18 @@ IRGenModule::getAddrOfLLVMVariable(LinkEntity entity,
   ForDefinition_t forDefinition = (ForDefinition_t) (definitionType != nullptr);
   LinkInfo link = LinkInfo::get(*this, entity, forDefinition);
 
-  // Clang may have defined the variable already.
-  if (auto existing = Module.getNamedGlobal(link.getName()))
+  // Clang may have defined the variable already, or we're referencing an entity
+  // which is known by IRGen, such as ().
+  if (auto existing = Module.getNamedGlobal(link.getName())) {
+    // If we're looking for ()'s type metadata, just return IRGen's already
+    // created global variable. We want the link entity to mangle for the
+    // address point, but ()'s metadata is the full metadata type, so the
+    // element bit cast is wrong for both address types.
+    if (entity.isTypeMetadata(Context.TheEmptyTupleType))
+      return existing;
```

Comment:
I feel like the special cases here are definitely being added at the wrong level.

Reply:
How would you suggest go about this? If I ask for LinkEntity's FullMetadata address point, I get the correct global variable here, but it mangles to Mf vs. N. I could alter LinkEntity's mangling to check for these types if that makes more sense.

Reply:
Well, let's see if this is actually necessary.  If this becomes a general conformance on tuple types, we'll need to be able to record that kind of specially, and the usefulness of directly linking to metadata will go away.